### PR TITLE
Explicit eager override

### PR DESCRIPTION
**Pull request by arnodenuijl**
Implementation for #34 

Shoot!

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -34,9 +34,11 @@ follow a structure similar to superagent's request API itself.
...
 - `redirects` *(Number)*: number of redirects to follow.
 - `eager` *(Boolean)*: whether or not to execute the request regardless of
-  usage of its corresponding response. Default value is `false` (i.e.,
-  the request is lazy). Main use case is: set this option to `true` if you
-  send POST requests and you are not interested in its response.
+  usage of its corresponding response. By default the eager setting of the
+  driver is used (whose default is `false`, i.e. the request is lazy). 
+  Explicitely setting eager in the request always overrides the driver 
+  setting. Main use case is: set this option to `true` if you send POST requests and
+  you are not interested in its response.
```
Comment (staltz) [inline docs/api.md]: Can I ask you to move these changes out of this file and into src/index.js? This file is auto-generated from that, so we never should directly modify this one.

File: src/http-driver.js [hunk 2]
```diff
@@ -128,7 +128,7 @@ function makeHTTPDriver({eager = false} = {eager: false}) {
       .map(request => {
         const reqOptions = normalizeRequestOptions(request)
         let response$ = createResponse$(reqOptions)
-        if (eager || reqOptions.eager) {
+        if (reqOptions.hasOwnProperty(`eager`) ? reqOptions.eager : eager) {
```
Comment (staltz) [inline src/http-driver.js]: Can we replace `hasOwnProperty` with `typeof reqOptions.eager === 'boolean'`? I suspect the latter is a tiny bit faster.

File: test/node.js [hunk 3]
```diff
@@ -32,6 +32,24 @@ describe('HTTP Driver in Node.js', function () {
...
+      });
+      var httpDriver = makeHTTPDriver({eager: true});
+      globalSandbox.petPOSTResponse = null;
+      httpDriver(request$);
+      setTimeout(function () {
+        assert.strictEqual(globalSandbox.petPOSTResponse, null);
+        done();
+      }, 100);
+    }
+  );
```
Comment (staltz) [inline test/node.js]: :+1:

File: test/node.js [hunk 4]
```diff
@@ -85,4 +103,25 @@ describe('HTTP Driver in Node.js', function () {
...
+      globalSandbox.petPOSTResponse = null;
+      httpDriver(request$);
+      setTimeout(function () {
+        assert.notStrictEqual(globalSandbox.petPOSTResponse, null);
+        assert.strictEqual(globalSandbox.petPOSTResponse, 'added Woof the Dog');
+        globalSandbox.petPOSTResponse = null;
+        done();
+      }, 100);
+    }
+  );
```
Comment (staltz) [inline test/node.js]: :+1:

File: docs/api.md (hunk 1 above)
Comment (arnodenuijl) [inline docs/api.md]: Done

File: src/http-driver.js (hunk 2 above)
Comment (arnodenuijl) [inline src/http-driver.js]: done

**Comment by staltz** [comment]
Thanks!

**Pull request merged**